sbin/woeusb

```diff
@@ -492,6 +494,23 @@ process_commandline_parameters(){
                     <<< "${parameters[0]}"
                 )"
                 ;;
+            --uefi-ntfs-img \
+            |-u)
+                shift_array parameters
+                uefi_ntfs_img="${parameters[0]}"
```

Comment:
We require all variables to be declared in the `local` scope whenever possible, use the name reference feature to allow the variable be modified in a function.

Reply:
> use the name reference feature to allow the variable be modified in a function.

Unfortunately, I have no idea what you mean by _name reference feature_.

In order to comply with your _all variables have to be local_ rule I declared the `$uefi_ntfs_img` variable in the `init()` function (line 64 in the following commit).

Reply:
> Unfortunately, I have no idea what you mean by name reference feature.

Refer:

* [Shell Parameters (Bash Reference Manual)](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameters.html)
* [Bash Builtins (Bash Reference Manual)](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html)
* [Short Detour into Namerefs | What's New in Bash Parameter Expansion | Linux Journal](https://www.linuxjournal.com/content/whats-new-bash-parameter-expansion#short-detour-into-namerefs)

> A variable can be assigned the nameref attribute using the -n option to the `declare` or `local` builtin commands (see [Bash Builtins](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#Bash-Builtins)) to create a nameref, or a reference to another variable. This allows variables to be manipulated indirectly. Whenever the nameref variable is referenced, assigned to, unset, or has its attributes modified (other than using or changing the nameref attribute itself), the operation is actually performed on the variable specified by the nameref variable’s value. A nameref is commonly used within shell functions to refer to a variable whose name is passed as an argument to the function. For instance, if a variable name is passed to a shell function as its first argument, running
>
>     declare -n ref=$1
>
> inside the function creates a nameref variable ref whose value is the variable name passed as the first argument. References and assignments to ref, and changes to its attributes, are treated as references, assignments, and attribute modifications to the variable whose name was passed as `$1`.

This feature allows a function to modify the value of the caller's variables (in this case the function is `        process_commandline_parameters` and the caller is `init`)  Note that nameref variables should be named with the `_ref` suffix as bash can't handle nameref variable has the same name with the variable it references.